tasks/libs/junit_upload.py

```diff
@@ -252,43 +258,40 @@ def repack_macos_junit_tar(infile, outfile):
             outfp.add(os.path.join(tempd, f), arcname=f)
 
 
-def retrieve_jira_card(xmlfile_path):
+def retrieve_jira_card(test_name, jira_project, jira_cache):
     """
-    Search in jira if a card already exist for the given testsuite
+    Search in jira if a card already exist for the given test
     """
-    from jira import JIRA
+    if test_name in jira_cache:
+        return jira_cache[test_name]
 
-    fake_jira_card = "XYZ-123"
-    tree = ET.parse(xmlfile_path)
+    jira_card = "XYZ-123"
```

Comment:
If I understand correctly, `XYZ-123` is returned only when we encounter an error somewhere. Could we return something like `ERROR: <info about error>` and parse the `ERROR` in the worklow, so we know we had an error and we also have some information about the error in the workflow? WDYT?

Reply:
We can have 2 kind of errors:
- jira token not found (should not happen)
- any issue using the JIRA api
as a consequence I can maybe send `ERROR-TOKEN` or `ERROR-API` as jira card name to adapt my message in the workflow, wdyt?

Reply:
Yes, sounds good! This way it also more explicit in the code what this jira_card value refers to!